torch/csrc/jit/passes/quantization.cpp

```diff
@@ -962,76 +954,109 @@ void InsertPrepackUnpack(script::Module& module) {
+        %w_quant = aten::quantize_per_channel(%w, %w_scale, %w_zero_point, %w_axis, %w_dtype)
+        %packed_params = quantized::linear_prepack(%w_quant, %b)
+        return (%packed_params) )";
+
   std::string conv2d_prepack = R"(
 graph(%a_dequant, %w, %b, %w_scale, %w_zero_point, %w_dtype, %stride, %padding, %dilation, %groups):
         %w_quant = aten::quantize_per_tensor(%w, %w_scale, %w_zero_point, %w_dtype)
         %packed_params = quantized::conv2d_prepack(%w_quant, %b, %stride, %padding, %dilation, %groups)
         return (%packed_params))";
 
-  // (is_conv, pattern, packed_params_module)
-  auto pattern_and_modules = {
-      std::make_tuple(false, linear_prepack, linear_params_module),
-      std::make_tuple(true, conv2d_prepack, conv_params_module)};
-  for (const auto& item : pattern_and_modules) {
-    bool is_conv = std::get<0>(item);
-    const std::string& pattern = std::get<1>(item);
-    const script::Module& packed_params_module = std::get<2>(item);
+  std::string conv2d_prepack_per_channel = R"(
+graph(%a_dequant, %w, %b, %w_scale, %w_zero_point, %w_axis, %w_dtype, %stride, %padding, %dilation, %groups):
+        %w_quant = aten::quantize_per_channel(%w, %w_scale, %w_zero_point, %w_axis, %w_dtype)
+        %packed_params = quantized::conv2d_prepack(%w_quant, %b, %stride, %padding, %dilation, %groups)
+        return (%packed_params))";
+
+  // (is_conv, is_per_channel, pattern, packed_params_module)
+  std::vector<PatternsAndModules> pattern_and_modules = {
+    {false, false, linear_prepack_per_tensor, linear_params_module},
+    {false, true, linear_prepack_per_channel, linear_params_module},
+    {true, false, conv2d_prepack, conv_params_module},
+    {true, true, conv2d_prepack_per_channel, conv_params_module}};
+  for (const auto& pm : pattern_and_modules) {
     Graph pattern_graph;
     std::unordered_map<std::string, Value*> vmap;
-    script::parseIR(pattern, &pattern_graph, vmap);
+    script::parseIR(pm.pattern, &pattern_graph, vmap);
     const auto& matches = findPatternMatches(pattern_graph, *graph);
     TORCH_INTERNAL_ASSERT(
         matches.size() <= 1, "We only support at most one match right now");
     for (const auto& match : matches) {
       const auto& match_vmap = match.values_map;
-      auto w_scale =
-          toIValue(match_vmap.at(vmap.at("w_scale"))).value().toDouble();
-      auto w_zero_point =
-          toIValue(match_vmap.at(vmap.at("w_zero_point"))).value().toInt();
-      auto w_dtype =
-          toIValue(match_vmap.at(vmap.at("w_dtype"))).value().toScalarType();
+      auto w_dtype_opt = getIValue("w_dtype", match_vmap, vmap);
+      auto w_scale_opt = getIValue("w_scale", match_vmap, vmap);
+      auto w_zero_point_opt = getIValue("w_zero_point", match_vmap, vmap);
+      if (!w_dtype_opt || !w_scale_opt || !w_zero_point_opt) {
+        GRAPH_DEBUG("dtype, scale or zero_point for weight is not constant, skipping the match.");
```

Comment:
Thanks for adding the logging! To make it more helpful it's better to add more details to the message: e.g. something to locate these values in the actual graph. Also, usually we print the graph before or after transformation with `GRAPH_DUMP` and report mutations we did on the graph with `GRAPH_UPDATE`.